test/integration/nodejs/nodejs_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022-2024 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package nodejs
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/parca-dev/parca-agent/pkg/agent"
+	"github.com/parca-dev/parca-agent/pkg/logger"
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+	"github.com/parca-dev/parca-agent/pkg/profiler/cpu"
+	"github.com/parca-dev/parca-agent/test/integration"
+)
+
+func TestNodeJS(t *testing.T) {
+	ok, _, err := agent.PreflightChecks(false, false, false)
+	require.Truef(t, ok, "preflight checks failed: %v", err)
+	if err != nil {
+		t.Logf("preflight checks passed but with errors: %v", err)
+	}
+
+	tests := []struct {
+		images  map[string]string
+		program string
+		want    []string
+		wantErr bool
+	}{
+		{
+			images: map[string]string{
+				"18": "18-slim",
+				"20": "20-slim",
+				"21": "21-slim",
+			},
+			program: "testdata/cpu_hog.js",
+			want: []string{
+				"JS:~",
+				"JS:~executeUserEntryPoint",
+				"JS:~Module._load",
+				"JS:~Module.load",
+				"JS:~Module._extensions..js",
+				"JS:~Module._compile",
+				"JS:~",
+				"JS:*runLoop",
```

Comment:
Maybe `NodeJS` is inlining these functions. Could we somehow disable inlining for the tests? Or a specific function?

Reply:
OK

Reply:
There is not a single option to disable inlining optimizations, I am using `--no-opt` to disable all optimizations. 
Now I can see the expected whole stack.

Reply:
@kakkoyun , please take another look